crates/modelardb_compression/src/lib.rs

```diff
@@ -309,60 +384,18 @@ impl<'a> CompressedSegmentBuilder<'a> {
 
         self.swing_could_fit_all =
             self.swing_could_fit_all && self.swing.fit_data_point(timestamp, value);
-
-        // Gorilla uses lossless compression and cannot exceed the error bound.
-        if self.gorilla.length < GORILLA_MAXIMUM_LENGTH {
-            self.gorilla.compress_value(value);
-        }
     }
 
     /// Return [`true`] if any of the current models can represent additional
     /// values, otherwise [`false`].
     fn can_fit_more(&self) -> bool {
-        self.pmc_mean_could_fit_all
-            || self.swing_could_fit_all
-            || self.gorilla.length < GORILLA_MAXIMUM_LENGTH
+        self.pmc_mean_could_fit_all || self.swing_could_fit_all
     }
 
-    /// Store the model that requires the smallest number of bits per value in
-    /// `compressed_record_batch_builder`. Returns the index of the first value
-    /// in `uncompressed_values` the selected model could not represent.
-    fn finish(self, compressed_record_batch_builder: &mut CompressedSegmentBatchBuilder) -> usize {
+    /// Return the model that requires the smallest number of bits per value if used for a segment.
+    fn finish(self) -> SelectedModel {
         // The model that uses the fewest number of bytes per value is selected.
```

Comment:
Maybe remove this comment since this is already explained in the methods doc comment above.

Reply:
Fixed.